code/modules/reagents/reagent_containers/syringes.dm

```diff
@@ -232,9 +235,35 @@
 		add_overlay(filling_overlay)
 	else
 		rounded_vol = 0
-
 	icon_state = "[rounded_vol]"
 	item_state = "syringe_[rounded_vol]"
+	if(ismob(loc))
+		var/injoverlay
+		switch(mode)
+			if (SYRINGE_DRAW)
+				injoverlay = "draw"
+			if (SYRINGE_INJECT)
+				injoverlay = "inject"
+		add_overlay(injoverlay)
+		update_wear_icon()
+
+/obj/item/weapon/reagent_containers/syringe/large/update_icon(A)
```

Comment:
You need to modify original ``update_icon`` to accept it's child object properly, no need to copypaste it here. Just refer to original object name from var in all icon state changes. Also never put procs for objects above their declaration.

Reply:
Please don't keep commented out code in a codebase.